community/cypher/cypher-compiler-2.2/src/main/scala/org/neo4j/cypher/internal/compiler/v2_2/planner/logical/idp/IDPSolver.scala

```diff
@@ -37,14 +37,14 @@ trait IDPSolverMonitor {
  *
  * written by Donald Kossmann and Konrad Stocker
  */
-class IDPSolver[S, P](generator: IDPSolverStep[S, P], // generates candidates at each step
-                      projectingSelector: ProjectingSelector[P], // pick best from a set of candidates
-                      registryFactory: () => IdRegistry[S] = () => IdRegistry[S], // maps from Set[S] to BitSet
-                      tableFactory: (IdRegistry[S], Seed[S, P]) => IDPTable[P] = (registry: IdRegistry[S], seed: Seed[S, P]) => IDPTable(registry, seed),
-                      maxTableSize: Int, // limits computation effort by reducing result quality
-                      monitor: IDPSolverMonitor) {
-
-  def apply(seed: Seed[S, P], initialToDo: Set[S])(implicit context: LogicalPlanningContext): Iterator[(Set[S], P)] = {
+class IDPSolver[S, P, C](generator: IDPSolverStep[S, P, C], // generates candidates at each step
```

Comment:
Couldn't we use longer names for these generic types? I had to think a couple of rounds extra to figure out what was what